### PR TITLE
Fix csv output interaction with fullsolve multiapps/eigensolves. 

### LINKED ISSUE #21252: Postprocessor output for repeated full-solves is not correct.
## Bug Description
In a repeated full-solve multiapp solution the oldest postprocessor value is printed, while the current one should be. This is apparent in an eigenvalue problem where the postprocessor output differs from the actual console output.

## Steps to Reproduce
Run the following test: [test/tests/problems/eigen_problem/eigensolvers/ne_coupled_picard_subT.i](https://github.com/idaholab/moose/pull/20269/files#diff-cdcc47d7b7723affe66b5023056b4c9c14cd4d4d53f448fdfd1a1c45e6f3745c) and observe the console and csv outputs.

## Impact
Weird results in postprocessing, increases user errors in reporting results if somebody relies on the csv output alone.

### PR DESCRIPTION
**Pull request by grmnptr**
Closes #21252

### CODE REVIEW

File: test/tests/multiapps/picard_postprocessor/steady_main.i [hunk 1]
```diff
@@ -78,6 +78,7 @@
     input_files = steady_sub.i
     clone_parent_mesh = true
     execute_on = 'timestep_begin'
+    keep_full_output_history = true
```
Comment (GiudGiud) [inline test/tests/multiapps/picard_postprocessor/steady_main.i:81]: what are you trying to preserve by adding this?
leave a comment for future maintainers

**Review by GiudGiud — COMMENTED** [review]
lgtm

File: test/tests/multiapps/picard_postprocessor/steady_main.i (hunk 1 above)
Comment (grmnptr) [reply test/tests/multiapps/picard_postprocessor/steady_main.i:82]: Add comments to all of the postprocessor input files!

File: test/tests/multiapps/secant_postprocessor/steady_main.i [hunk 2]
```diff
@@ -77,6 +77,8 @@
     input_files = steady_sub.i
     clone_parent_mesh = true
     execute_on = 'timestep_begin'
+    # we keep the full potprocessor output history of the subapp
```
Comment (GiudGiud) [inline test/tests/multiapps/secant_postprocessor/steady_main.i:80]:
```suggestion
    # we keep the full postprocessor output history of the subapp
```

File: test/tests/multiapps/steffensen_postprocessor/steady_main.i [hunk 3]
```diff
@@ -77,6 +77,8 @@
     input_files = steady_sub.i
     clone_parent_mesh = true
     execute_on = 'timestep_begin'
+    # we keep the full potprocessor output history of the subapp
```
Comment (GiudGiud) [inline test/tests/multiapps/steffensen_postprocessor/steady_main.i:80]:
```suggestion
    # we keep the full postprocessor output history of the subapp
```

**Review by GiudGiud — COMMENTED** [review]
this doesnt really explain why you are doing it. Is the formatter closing the file when it used not to for subapps?

**Comment by GiudGiud** [comment]
Let's squash those last 3 commits together they are 3 lines

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**